node/src/accountant/payment_adjuster/mod.rs

```diff
@@ -0,0 +1,2340 @@
+            extreme_payables,
+            &PRESERVED_TEST_PAYMENT_THRESHOLDS,
+            now,
+        );
+        let mut subject = PaymentAdjusterReal::new();
+        subject.logger = Logger::new(test_name);
+        // In turn, tiny cw balance
+        let cw_service_fee_balance = 1_000;
+        let agent = {
+            let mock = BlockchainAgentMock::default()
+                .service_fee_balance_minor_result(cw_service_fee_balance);
+            Box::new(mock)
+        };
+        let adjustment_setup = PreparedAdjustment {
+            agent,
+            adjustment_analysis: AdjustmentAnalysis::new(
+                Adjustment::ByServiceFee,
+                analyzed_payables,
+            ),
+            response_skeleton_opt: None,
+        };
+
+        let result = subject.adjust_payments(adjustment_setup, now);
+
+        // The error isn't important. Received just because we set an almost empty wallet
+        let err = match result {
+            Ok(_) => panic!("we expected err but got ok"),
+            Err(e) => e,
+        };
+        assert_eq!(err, PaymentAdjusterError::AllAccountsEliminated);
+        let expected_log = |wallet: &str, proposed_adjusted_balance_in_this_iteration: u64| {
+            format!(
+                "INFO: {test_name}: Shortage of MASQ in your consuming wallet will impact payable \
+                {wallet}, ruled out from this round of payments. The proposed adjustment {} wei was \
+                below the disqualification limit {} wei",
+                proposed_adjusted_balance_in_this_iteration.separate_with_commas(),
+                (*MAX_POSSIBLE_SERVICE_FEE_BALANCE_IN_MINOR).separate_with_commas()
+            )
+        };
+        let log_handler = TestLogHandler::new();
+        // Notice that the proposals grow as one disqualified account drops out in each iteration
+        log_handler.exists_log_containing(&expected_log(
+            "0x000000000000000000000000000000626c616830",
+            333,
+        ));
+        log_handler.exists_log_containing(&expected_log(
+            "0x000000000000000000000000000000626c616831",
+            499,
+        ));
+        log_handler.exists_log_containing(&expected_log(
+            "0x000000000000000000000000000000626c616832",
+            999,
+        ));
+    }
+
+    fn meaningless_timestamp() -> SystemTime {
+        SystemTime::now()
+    }
+
+    // This function should take just such essential args as balances and those that play rather
+    // a secondary role, yet an important one in the verification processes for proposed adjusted
+    // balances. Refrain from employing more of the weights-affecting parameters as they would
+    // only burden us with their consideration in these tests.
+    fn make_plucked_qualified_account(
+        wallet_addr_fragment: &str,
+        balance_minor: u128,
+        threshold_intercept_major: u128,
+        permanent_debt_allowed_major: u128,
```

Comment:
An amazing idea would be if you pass the `disqualification_limit` instead of `threshold_intercept_major` and `permanent_debt_allowed_major`, and you'll compute these values inside the function.